### PR TITLE
ARROW-11565: [C++][Gandiva] Modify upper()/lower() to work with UTF8 and add INIT_CAP function

### PR DESCRIPTION
**Pull request by anthonylouisbsb**
It finishes the implementation that started in the https://github.com/apache/arrow/pull/9450 pull request

### CODE REVIEW

**Comment by anthonylouisbsb** [comment]
@projjal The [JNI build](https://github.com/apache/arrow/pull/10040/checks?check_run_id=2352789853) is always failing with a message that the LLVM dependency could not be found. I notice that the same build is not failing in other Pull Requests.

I only added the **utf8proc** dependency inside the Cmake files. I did not remove any dependency from CmakeLists.txt, so do you have any idea why only this building is failing?

**Comment by anthonylouisbsb** [comment]
@projjal can you close the https://github.com/apache/arrow/pull/9450 Pull Request?

File: cpp/src/gandiva/gdv_function_stubs.cc [hunk 1]
```diff
@@ -1000,5 +1172,47 @@ void ExportedStubFunctions::AddMappings(Engine* engine) const {
   engine->AddGlobalMappingForFunc("gdv_fn_sha256_decimal128",
                                   types->i8_ptr_type() /*return_type*/, args,
                                   reinterpret_cast<void*>(gdv_fn_sha256_decimal128));
+
+  // gdv_fn_utf8_char_length
+  args = {
```
Comment (projjal) [inline cpp/src/gandiva/gdv_function_stubs.cc:1177]: you don't need to register these here. only upper/lower is enough.

File: ci/scripts/python_wheel_manylinux_build.sh [hunk 2]
```diff
@@ -64,6 +64,7 @@ echo "=== (${PYTHON_VERSION}) Building Arrow C++ libraries ==="
 : ${ARROW_WITH_LZ4:=ON}
 : ${ARROW_WITH_SNAPPY:=ON}
 : ${ARROW_WITH_ZLIB:=ON}
+: ${ARROW_WITH_UTF8PROC:=ON}
```
Comment (projjal) [inline ci/scripts/python_wheel_manylinux_build.sh:67]: Are these necessary?

File: cpp/src/gandiva/gdv_function_stubs.cc (hunk 1 above)
Comment (anthonylouisbsb) [reply cpp/src/gandiva/gdv_function_stubs.cc]: Functions removed from global mapping.

File: ci/scripts/python_wheel_manylinux_build.sh (hunk 2 above)
Comment (anthonylouisbsb) [reply ci/scripts/python_wheel_manylinux_build.sh]: No, in the CmakeOptions.cmake we set the variable if the Gandiva or Kernel is computed.

**Comment by anthonylouisbsb** [comment]
@projjal corrections made and I added the changes from the https://github.com/apache/arrow/pull/10193 PR

**Review by projjal — APPROVED** [review]

**Comment by projjal** [comment]
> This PR is related to the https://issues.apache.org/jira/browse/ARROW-12592 issue too

Add the ticket in the PR title too

**Comment by projjal** [comment]
> @projjal how is the format of the title with two tickets? Would it be `ARROW-11565: ARROW-12592: [C++][Gandiva] Modify upper()/lower() logic and add INIT_CAP function`?
> 
> Or would be better to close the ticket in Jira and mark it as duplicated, pointing to the ticket of the upper and lower function?

"ARROW-11565, ARROW-12592: " should be fine. Don't know if the github bot will pick up the jira ticket from the description. If not please comment & close the jira ticket when this gets merged.

**Comment by anthonylouisbsb** [comment]
@projjal I changed the https://issues.apache.org/jira/browse/ARROW-11565 ticket title and I closed the https://issues.apache.org/jira/browse/ARROW-12592 ticket marking as duplicate

**Review by projjal — APPROVED** [review]

**Pull request closed without merging**